components/camel-pg-replication-slot/src/main/java/org/apache/camel/component/pg_replication_slot/PgReplicationSlotConsumer.java

```diff
@@ -0,0 +1,257 @@
+        if (this.connection != null) {
+            this.connection.close();
+            this.connection = null;
+        }
+
+        if (this.scheduledExecutor != null) {
+            this.getEndpoint().getCamelContext().getExecutorServiceManager().shutdownNow(this.scheduledExecutor);
+            this.scheduledExecutor = null;
+        }
+    }
+
+    @Override
+    protected int poll() throws Exception {
+        PGReplicationStream stream = getStream();
+
+        if (stream == null) {
+            return 0;
+        }
+
+        try {
+            // The same payload will be sent again and again until the processing is completed successfully.
+            // We should not read another payload before that to guarantee the order of processing.
+            if (this.payload == null) {
+                ByteBuffer msg = stream.readPending();
+
+                if (msg == null) {
+                    return 0;
+                }
+
+                int offset = msg.arrayOffset();
+                byte[] source = msg.array();
+                int length = source.length - offset;
+
+                this.payload = new byte[length];
+                System.arraycopy(source, offset, this.payload, 0, length);
+            }
+        } catch (PSQLException e) {
+            validateConnection(e);
+            throw e;
+        }
+
+        Exchange exchange = this.endpoint.createExchange();
+        exchange.setExchangeId(stream.getLastReceiveLSN().asString());
+
+        Message message = exchange.getIn();
+        message.setBody(this.payload);
+
+        final long delay = this.endpoint.getStatusInterval();
+        ScheduledFuture<?> scheduledFuture = this.scheduledExecutor.scheduleAtFixedRate(() -> {
+            try {
+                log.debug("Processing took too long. Sending status update to avoid disconnect.");
+                stream.forceUpdateStatus();
+            } catch (SQLException e) {
+                log.error(e.getMessage(), e);
+            }
+        }, delay, delay, TimeUnit.SECONDS);
+
+        exchange.addOnCompletion(new Synchronization() {
+            @Override
+            public void onComplete(Exchange exchange) {
+                processCommit(exchange);
+                scheduledFuture.cancel(true);
+            }
+
+            @Override
+            public void onFailure(Exchange exchange) {
+                processRollback(exchange);
+                scheduledFuture.cancel(true);
+            }
+        });
+
+        getProcessor().process(exchange);
+
+        return 1;
+    }
+
+    private void processCommit(Exchange exchange) {
+        try {
+            PGReplicationStream stream = getStream();
+
+            if (stream == null) {
+                return;
+            }
+
+            stream.setAppliedLSN(stream.getLastReceiveLSN());
+            stream.setFlushedLSN(stream.getLastReceiveLSN());
+            stream.forceUpdateStatus();
+
+            this.payload = null;
+        } catch (PSQLException e) {
```

Comment:
single catch with | notation can be an option. both exception seems closer, may be good to reconsider this part to simplify and save a couple if lines.